src/daemon/rpmostree-sysroot-upgrader.cxx

```diff
@@ -450,27 +450,14 @@ rpmostree_sysroot_upgrader_pull_base (RpmOstreeSysrootUpgrader  *self,
         if (override_commit)
           return glnx_throw (error, "Specifying commit overrides for container-image-reference type refspecs is not supported");
 
-        if (strstr (refspec, "@"))
-          return glnx_throw (error, "Pinned to specific container image digest, cannot upgrade");
-
-        /* Check to see if the digest that `refspec` points to is the same before pulling a new container image. */
-        const char *cur_digest = rpmostree_origin_get_container_image_reference_digest (self->original_origin);
-        if (cur_digest)
+        auto refspec_s = std::string(refspec);
+        auto import = rpmostreecxx::pull_container(*self->repo, *cancellable, refspec_s);
```

Comment:
`CXX_TRY_VAL`?

Reply:
Thanks, I will try to remember that.  Ideally we'd have a static analysis check, but that seems nontrivial.